src/main/java/org/opensearch/neuralsearch/search/query/HybridQueryPhaseSearcher.java

```diff
@@ -36,6 +40,14 @@
 @Log4j2
 public class HybridQueryPhaseSearcher extends QueryPhaseSearcherWrapper {
 
+    private final QueryPhaseSearcher defaultQueryPhaseSearcherWithEmptyCollectorContext;
```

Comment:
Seems like it's better to use `QueryPhaseSearcherWrapper` as composition instead of inheritance. We can add a constructor in `QueryPhaseSearcherWrapper` to accept both`QueryPhase.DefaultQueryPhaseSearcher` and `ConcurrentQueryPhaseSearcher`. Then `HybridQueryPhaseSearcher` can have two `QueryPhaseSearcherWrapper` instances, one is `defaultQueryPhaseSearcherWrapper`, the other as one is `queryPhaseSearcherWrapperWithEmptyCollectorContext`, then in `searchWith`, it's either `defaultQueryPhaseSearcherWrapper.searchWith` or `queryPhaseSearcherWrapperWithEmptyCollectorContext.searchWith`

Reply:
For this we need change in OS core, but owners from core side are against it, method override is their preferred approach, mainly to keep changes on core side to minimum. You can check the discussion details in corresponding core PR https://github.com/opensearch-project/OpenSearch/issues/13170

Reply:
The only change on core side from my suggestion is to add an additional constructor so that client can provide default and concurrent QueryPhaseSearcher. I'm afraid it's not the solution #13170 describes and core team is against.